domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt

```diff
@@ -261,37 +346,387 @@ class TopicListController @Inject constructor(
+    if (story.chapterList.size > nextChapterIndex) {
+      val nextChapterSummary: ChapterSummary? = story.chapterList[nextChapterIndex]
+      if (nextChapterSummary != null) {
+        return createPromotedStory(
+          storyId,
+          topic,
+          completedChapterProgressList.size,
+          story.chapterCount,
+          nextChapterSummary.name,
+          nextChapterSummary.explorationId,
+          isTopicConsideredCompleted
+        )
+      }
+    }
+    return null
+  }
+
+  private fun ChapterProgress.getNumberOfDaysPassed(): Long {
+    return TimeUnit.MILLISECONDS.toDays(
+      oppiaClock.getCurrentCalendar().timeInMillis - this.lastPlayedTimestamp
+    )
+  }
+
+  // TODO(#2550): Remove hardcoded order of topics. Compute list of suggested stories from backend structures
+  /** Returns a list of topic IDs for which the specified topic ID expects to be completed before being suggested.
+   * Dependent hierarchy
+   * TEST_TOPIC_ID_0 (Fractions)
+   * TEST_TOPIC_ID_1 (TEST_TOPIC_ID_0,Ratios)
+   * Addition and Subtraction (depends on Place Values)
+   * Multiplication (depends on Addition and Subtraction)
+   * Division (depends on Multiplication)
+   * Expressions and Equations (depends on A+S, Multiplication, Division)
+   * Fractions (depends on A+S, Multiplication, Division)
+   * Ratios (depends on A+S, Multiplication, Division)
+   * Decimals (depends on A+S, Multiplication, Division)
+   *
+   * */
+  private fun retrieveTopicDependencies(topicId: String): List<String> {
+    val listOfTopicIds = mutableListOf<String>()
+    when (topicId) {
+      TEST_TOPIC_ID_0 -> {
+        listOfTopicIds.add(FRACTIONS_TOPIC_ID)
+      }
+      TEST_TOPIC_ID_1 -> {
+        listOfTopicIds.add(TEST_TOPIC_ID_0)
+        listOfTopicIds.add(RATIOS_TOPIC_ID)
+      }
+      FRACTIONS_TOPIC_ID -> {
+        listOfTopicIds.add(ADDITION_AND_SUBTRACTION_TOPIC_ID)
+        listOfTopicIds.add(MULTIPLICATION_TOPIC_ID)
+        listOfTopicIds.add(DIVISION_TOPIC_ID)
+      }
+      RATIOS_TOPIC_ID -> {
+        listOfTopicIds.add(ADDITION_AND_SUBTRACTION_TOPIC_ID)
+        listOfTopicIds.add(MULTIPLICATION_TOPIC_ID)
+        listOfTopicIds.add(DIVISION_TOPIC_ID)
+      }
+      ADDITION_AND_SUBTRACTION_TOPIC_ID -> {
+        listOfTopicIds.add(PLACE_VALUE_TOPIC_ID)
+      }
+      MULTIPLICATION_TOPIC_ID -> {
+        listOfTopicIds.add(ADDITION_AND_SUBTRACTION_TOPIC_ID)
+      }
+      DIVISION_TOPIC_ID -> {
+        listOfTopicIds.add(MULTIPLICATION_TOPIC_ID)
+      }
+      EXPRESSION_AND_EQUATION_TOPIC_ID -> {
+        listOfTopicIds.add(ADDITION_AND_SUBTRACTION_TOPIC_ID)
+        listOfTopicIds.add(MULTIPLICATION_TOPIC_ID)
+        listOfTopicIds.add(DIVISION_TOPIC_ID)
+      }
+      DECIMALS_TOPIC_ID -> {
+        listOfTopicIds.add(ADDITION_AND_SUBTRACTION_TOPIC_ID)
+        listOfTopicIds.add(MULTIPLICATION_TOPIC_ID)
+        listOfTopicIds.add(DIVISION_TOPIC_ID)
+      }
+    }
+    return listOfTopicIds
```

Comment:
```suggestion
    return when (topicId) {
      TEST_TOPIC_ID_0 -> listOf(FRACTIONS_TOPIC_ID)
      TEST_TOPIC_ID_1 -> listOf(TEST_TOPIC_ID_0, RATIOS_TOPIC_ID)
      FRACTIONS_TOPIC_ID -> {
        listOf(ADDITION_AND_SUBTRACTION_TOPIC_ID, MULTIPLICATION_TOPIC_ID, DIVISION_TOPIC_ID)
      }
      RATIOS_TOPIC_ID ->  {
        listOf(ADDITION_AND_SUBTRACTION_TOPIC_ID, MULTIPLICATION_TOPIC_ID, DIVISION_TOPIC_ID)
      }
      ADDITION_AND_SUBTRACTION_TOPIC_ID -> listOf(PLACE_VALUE_TOPIC_ID)
      MULTIPLICATION_TOPIC_ID -> listOf(ADDITION_AND_SUBTRACTION_TOPIC_ID)
      DIVISION_TOPIC_ID -> listOf(MULTIPLICATION_TOPIC_ID)
      EXPRESSION_AND_EQUATION_TOPIC_ID ->  {
        listOf(ADDITION_AND_SUBTRACTION_TOPIC_ID, MULTIPLICATION_TOPIC_ID, DIVISION_TOPIC_ID)
      }
      DECIMALS_TOPIC_ID ->  {
        listOf(ADDITION_AND_SUBTRACTION_TOPIC_ID, MULTIPLICATION_TOPIC_ID, DIVISION_TOPIC_ID)
      }
      else -> listOf()
    }
```

Rearranging to avoid a variable & be a bit more Kotlin-idiomatic.